src/user/user.service.ts

```diff
@@ -0,0 +1,93 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { CreateUserDTO } from './dto/create-user.dto';
+import { PatchUserDTO } from './dto/patch-user.dto';
+import { GithubProfileService } from './github-profile.service';
+import { UserModel } from './schema/user.schema';
+
+@Injectable()
+export class UserService {
+  constructor(
+    private readonly githubProfileService: GithubProfileService,
+    @InjectModel(UserModel.name) private readonly userModel: Model<UserModel>,
+  ) {}
+
+  public async create(userDTO: CreateUserDTO) {
+    let githubProfile = null;
+    if (userDTO.githubUsername) {
+      githubProfile = await this.githubProfileService.create(
+        userDTO.githubUsername,
+      );
+    }
+
+    const createdUser = new this.userModel({
+      _id: userDTO.discordUsername,
+      avatar: userDTO.avatar,
+      type: userDTO.type,
+      bio: userDTO.bio,
+      github: githubProfile,
+    });
+
+    try {
+      return await createdUser.save();
+    } catch (error) {
+      throw new HttpException(error.message, HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  public async patch(userDTO: PatchUserDTO, id: string) {
+    let githubProfile = null;
+    if (userDTO.githubUsername) {
+      githubProfile = await this.githubProfileService.create(
```

Comment:
This needs to be adjusted. In our use-case we probably need to delete the embeded document from the collection and recreate it.